### PR TITLE
Toolbar refactor as class and an extra.

### PR DESCRIPTION
**Pull request by sposmen**
Hi @rogerpadilla, this is a refactor of the Toolbar as class to reuse some variables.
Next Webpack Uglifier plugin compatible with ECMA 6 to compress the code in the releases.

### CODE REVIEW

**Comment by sposmen** [comment]
@rogerpadilla ready the changes... also remember that now you can set the build to release it compressed.

File: config/webpack.prod.js [hunk 1]
```diff
@@ -31,8 +32,12 @@ module.exports = function (options) {
         }
       }),
       new webpack.NoEmitOnErrorsPlugin(),
-      new webpack.optimize.UglifyJsPlugin({
-        sourceMap: true
+      new UglifyJSPlugin({
```
Comment (rogerpadilla) [inline config/webpack.prod.js]: Why replacing the "UglifyJsPlugin" with this one? it looks like the one provided by webpack should do the work also.

File: src/app/toolbar/toolbar.ts [hunk 2]
```diff
@@ -1,182 +1,162 @@
...
-    const newCopiedRow = TcGridUtil.CopyRow(newRow, true, true);
-    newCopiedRow.WeekNumber = TcGridUtil.IdentifyWeekNumber(newCopiedRow.InDate);
+    rowsToProcess = rows.slice(srcIdx + 1);
 
-    const newPos = TcGridUtil.StoreItemLocation(rows, newIdx);
-    rows[newPos] = newCopiedRow;
+    rowsToProcess.forEach((row: any) => {
+      const date = row.InDate;
+      if (row.RecordType === TcGridUtil.RecordTypes.DatePlaceholder && checkIsWeekday(date)) {
+        _.assign(row, dataToClone);
```
Comment (rogerpadilla) [inline src/app/toolbar/toolbar.ts]: Use Object.assign in this case (prefer JavaScript native utilities when possible).

File: src/app/toolbar/toolbar.ts [hunk 3]
```diff
@@ -1,182 +1,162 @@
...
+  country: any;
+  countryCodeKey: string;
+  countryCode: string;
+  toolbarHtml: string;
+  appContainer: any;
+
+  constructor() {
+    // tslint:disable-next-line:no-require-imports
+    this.toolbarHtml = require('./toolbar.tpl');
+    this.appContainer = document.querySelector('#appContainer');
```
Comment (rogerpadilla) [inline src/app/toolbar/toolbar.ts]: This is only referenced in a single method, don't need to catch it in the state of the object.

File: src/app/toolbar/toolbar.ts [hunk 4]
```diff
@@ -1,182 +1,162 @@
...
+  copy: any;
+  country: any;
+  countryCodeKey: string;
+  countryCode: string;
+  toolbarHtml: string;
+  appContainer: any;
+
+  constructor() {
+    // tslint:disable-next-line:no-require-imports
+    this.toolbarHtml = require('./toolbar.tpl');
```
Comment (rogerpadilla) [inline src/app/toolbar/toolbar.ts]: This is only referenced in a single method, don't need to catch it in the state of the object.

File: src/app/toolbar/toolbar.ts [hunk 5]
```diff
@@ -1,182 +1,162 @@
...
-  const toolbar = document.querySelector('.adp-next');
-  const copy = toolbar.querySelector('.adp-next__copy');
-  const country = toolbar.querySelector('.adp-next__country') as HTMLSelectElement;
+    this.appContainer.insertAdjacentHTML('afterbegin', this.toolbarHtml);
 
-  const countryCodeKey = 'adp-next__countryCode';
+    this.toolbar = document.querySelector('.adp-next');
+    this.copy = this.toolbar.querySelector('.adp-next__copy');
+    this.country = this.toolbar.querySelector('.adp-next__country') as HTMLSelectElement;
+    this.countryCodeKey = 'adp-next__countryCode';
```
Comment (rogerpadilla) [inline src/app/toolbar/toolbar.ts]: This is not directly related to the dom, consider initializing it as a readonly variable in the same place you declare it.

File: config/webpack.prod.js (hunk 1 above)
Comment (sposmen) [reply config/webpack.prod.js]: ES6 support on build

File: src/app/toolbar/toolbar.ts (hunk 4 above)
Comment (sposmen) [reply src/app/toolbar/toolbar.ts]: The method could be called multiple times requiring it multiple times also.

File: src/app/toolbar/toolbar.ts (hunk 3 above)
Comment (sposmen) [reply src/app/toolbar/toolbar.ts]: Same as toolbarHtml... multiple calls multiple required calls.
